File: micro-common/src/main/java/com/jiangtj/micro/common/validation/MaxLengthValidator.java
package com.jiangtj.micro.common.validation;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

public class MaxLengthValidator implements ConstraintValidator<MaxLength, String> {

    private int max;
    private boolean trim;

    @Override
    public void initialize(MaxLength constraintAnnotation) {
        this.max = constraintAnnotation.value();
        this.trim = constraintAnnotation.trim();
    }

    @Override
    public boolean isValid(String value, ConstraintValidatorContext context) {
        if (value == null) {
            return true;
        }
        if (trim) {
            value = value.trim();
        }
        int length = value.length();
        return length <= max;
    }
}

File: micro-common/src/main/java/com/jiangtj/micro/common/validation/MaxLengthValidator.kt
package com.jiangtj.micro.common.validation

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

class MaxLengthValidator : ConstraintValidator<MaxLength, String?> {
    private var max = 0
    private var trim = false

    override fun initialize(constraintAnnotation: MaxLength) {
        this.max = constraintAnnotation.value
        this.trim = constraintAnnotation.trim
    }

    override fun isValid(value: String?, context: ConstraintValidatorContext?): Boolean {
        var value = value ?: return true
        if (trim) {
            value = value.trim()
        }
        val length = value.length
        return length <= max
    }
}
